python/pyiceberg/schema.py

```diff
@@ -337,6 +338,142 @@ def primitive(self, primitive: PrimitiveType) -> T:
         """Visit a PrimitiveType"""
 
 
+class SchemaWithPartnerVisitor(Generic[P, T], ABC):
+    def before_field(self, field: NestedField, field_partner: Optional[P]) -> None:
```

Comment:
Shouldn't we just omit those before/after methods? Since it is quite a hot path, I'd rather remove them if we don't need/use them?

Reply:
I'd probably keep them so that we don't have to translate them later.